### PR TITLE
Fix arguments of overridden `set_style` for `BackgroundRectangle`

### PR DESCRIPTION
**Pull request by dawnsqrl**
## Overview: What does this pull request change?

Fixes issue #2337:
Using `Write()` animation on a `Text` object with `.add_background_rectangle()` applied no longer generates a `TypeError`.

### **Before**
The overridden `set_style()` method for `BackgroundRectangle` used to accept only a reduced number of arguments:
```python
def set_style(
        self,
        stroke_color=None,
        stroke_width=None,
        fill_color=None,
        fill_opacity=None,
        family=True,
    ):
    ...
```
As a result, when the following scene is rendered, `manim` reports `TypeError: BackgroundRectangle.set_style() got an unexpected keyword argument 'stroke_opacity'`.
```python
class TextBackground(Scene):
    def construct(self):
        text = Text("Some text")
        text.add_background_rectangle(color=TEAL)
        w = Write(text)
        self.play(w)
        self.wait()
    ):
```
### **After**
The function signature of `set_style()` for `BackgroundRectangle` is now simplified using `kwargs`, while only `fill_opacity` is included explicitly. In case `kwargs` captures any other arguments, a relevant piece of info is output to log.
```python
def set_style(self, fill_opacity, **kwargs):
    super().set_style(
        stroke_color=BLACK,
        stroke_width=0,
        fill_color=BLACK,
        fill_opacity=fill_opacity,
    )
    if len(kwargs) > 0:
        logger.info(
            "set_style for BackgroundRectangle does not process arguments other than fill_opacity."
        )
    return self
```
This restores all possible argument dependencies when animating `BackgroundRectangle`. The test scene `TextBackground` now renders correctly with intended effects.

https://user-images.githubusercontent.com/35942162/161867007-99e00b96-496a-40bf-bb7b-18f3ae366065.mp4

## Links to added or changed documentation pages

The `set_style` method of `BackgroundRectangle` and `VMobject` both have empty documentation.

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

**Review by behackl — COMMENTED** [review]
Hello! Thank you for your contribution!

There is one problem I see with this fix, namely that it needlessly replicates information. Instead of introducing more keyword arguments, what do you think about removing all the ones that are not passed to super (or specified automatically in the recursive `(V)Mobject.set_style` calls like `family`), and using a catchall argument `**kwargs`? We could even go as far as to log a warning when the user wants to set properties that do not actually change (i.e., if `kwargs` is not empty).

**Comment by dawnsqrl** [comment]
Thank you for the response! I probably need to review the original version and see why only those specific arguments were included. The additional ones will probably get merged into `**kwargs`, if they are not strictly necessary in this reported bug case.

**Comment by behackl** [comment]
> Thank you for the response! I will need to review the original version and see why only those specific arguments were included. The additional ones will probably get merged into `**kwargs`, if they are not strictly necessary in this reported bug case.

If you look at the original function you will see that
```py
super().set_style(
    stroke_color=BLACK,
    stroke_width=0,
    fill_color=BLACK,
     fill_opacity=fill_opacity,
)
```
is how `BackgroundRectangle` propagates the `set_style` call, only the `fill_opacity` can actually be changed and all other values are hardcoded. I'd suggest rewriting the method signature as `def set_style(self, fill_opacity, **kwargs)` and then including a logger warning or info documenting that `BackgroundRectangle` did not process arguments in `kwargs`, if there are any.

**Comment by dawnsqrl** [comment]
The argument of `set_style` has been simplified and I used `logger.info()` to send out a notice when `kwargs` is not empty. As expected, when `Write(text)` is called some new INFO appears in the terminal, while using `Create(text)` instead triggers no additional INFO. This agrees with the description of the original issue reporter.

File: manim/mobject/geometry/shape_matchers.py [hunk 1]
```diff
@@ -104,21 +104,19 @@ def pointwise_become_partial(self, mobject, a, b):
...
+        # All other style arguments are ignored
         super().set_style(
             stroke_color=BLACK,
             stroke_width=0,
             fill_color=BLACK,
             fill_opacity=fill_opacity,
         )
+        if len(kwargs) > 0:
+            logger.info(
+                "set_style for BackgroundRectangle does not process arguments other than fill_opacity."
```
Comment (behackl) [inline manim/mobject/geometry/shape_matchers.py]:
```suggestion
                "Arguments %s are ignored in BackgroundRectangle.set_style.",
                kwargs
```

Just slightly more verbose output.

**Review by behackl — APPROVED** [review]
Thank you! I've pushed a commit that makes the logger output slightly more verbose and added a test -- could you take a look whether my commit looks right to you? If so, this can be merged. :-)

**Comment by dawnsqrl** [comment]
The newly added commit looks great to me! Thank you for the review!

**Pull request merged**